docs/client_advanced.rst

```diff
@@ -592,14 +594,18 @@ Authentication credentials can be passed in proxy URL::
 Contrary to the ``requests`` library, it won't read environment
 variables by default. But you can do so by passing
 ``trust_env=True`` into :class:`aiohttp.ClientSession`
-constructor for extracting proxy configuration from
-*HTTP_PROXY*, *HTTPS_PROXY*, *WS_PROXY* or *WSS_PROXY* *environment
-variables* (all are case insensitive)::
+constructor.::
 
    async with aiohttp.ClientSession(trust_env=True) as session:
        async with session.get("http://python.org") as resp:
            print(resp.status)
 
+.. note::
+    ``aiohttp`` uses `urllib.request.getproxies <https://docs.python.org/3/library/urllib.request.html#urllib.request.getproxies>`_
+    for reading the proxy configuration and applies them for the schemas ``http``, ``https``, ``ws`` and ``wss``
+
+    Hosts defined in ``no_proxy`` will bypass the proxy.
```

Comment:
If this is true, it should probably be documented in the Python docs. I couldn't find anything about no_proxy in our code or in the Python documentation last time I looked.

Reply:
`getproxies` is reading `.*_proxy` from the env var and therefore as well `no_proxy`
```
// getproxies ouptut
{'no': ',localhost, 'http': 'http://some-host.com'}
```
The code uses then the helper function [urllib.request.proxy_bypass](https://github.com/python/cpython/blob/3.11/Lib/urllib/request.py#L2643) at this [postion](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/helpers.py#L303) to check if the proxy shall be bypassed for a given host.
Unfortunaltey `urllib.request.proxy_bypass` isn't documneted like `getproxies` in the python docs.